### PR TITLE
Adding encoding function to listAllMails

### PR DESCRIPTION
**Pull request by ashalya**

### CODE REVIEW

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/messages/getAMail.xml [hunk 1]
```diff
@@ -25,6 +25,13 @@
         <property name="uri.var.format" expression="$func:format"/>
         <property name="uri.var.metadataHeaders" expression="$func:metadataHeaders"/>
         <property name="gmailParameters" value=""/>
+        <script language="js">
+            <![CDATA[
+                   var query = mc.getProperty('uri.var.metadataHeaders');
+					var encodedQuery = encodeURIComponent(query);
```
Comment (kesavany) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/messages/getAMail.xml]: Reformat the code.

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/messages/listAllMails.xml [hunk 2]
```diff
@@ -29,6 +29,13 @@
         <property name="uri.var.pageToken" expression="$func:pageToken"/>
         <property name="uri.var.q" expression="$func:q"/>
         <property name="gmailParameters" value=""/>
+        <script language="js">
+            <![CDATA[
+                   var query = mc.getProperty('uri.var.q');
+					var encodedQuery = encodeURIComponent(query);
+					mc.setProperty('uri.var.q', encodedQuery);
```
Comment (kesavany) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/messages/listAllMails.xml]: Reformat the code.

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/createMail.java [hunk 3]
```diff
@@ -33,8 +33,8 @@ public void connect(MessageContext messageContext) throws ConnectException {
             String parameter = messageContext.getProperty("parameters").toString();
             byte[] encodedBytes = Base64.encodeBase64(parameter.getBytes());
             String encodedVal=new String(encodedBytes);
-             encodedVal= encodedVal.replace('+', '-');
-             encodedVal=encodedVal.replace('/', '_');
+            encodedVal= encodedVal.replace('+', '-');
```
Comment (Shakila) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/createMail.java]: Need to format code

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/messages/getAMail.xml [hunk 4]
```diff
@@ -25,6 +25,13 @@
         <property name="uri.var.format" expression="$func:format"/>
         <property name="uri.var.metadataHeaders" expression="$func:metadataHeaders"/>
         <property name="gmailParameters" value=""/>
+        <script language="js">
+            <![CDATA[
+                   var query = mc.getProperty('uri.var.metadataHeaders');
```
Comment (Shakila) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/messages/getAMail.xml]: Use 4 spaces for indentation.

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/threads/getAThread.xml [hunk 5]
```diff
@@ -26,6 +26,11 @@
         <property name="uri.var.format" expression="$func:format"/>
         <property name="uri.var.metadataHeaders" expression="$func:metadataHeaders"/>
         <property name="gmailParameters" value=""/>
+        <![CDATA[
+                   var query = mc.getProperty('uri.var.metadataHeaders');
```
Comment (Shakila) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/threads/getAThread.xml]: Use 4 spaces for indentation. Do this in whole file.

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/createMail.java (hunk 3 above)
Comment (ashalya) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/createMail.java]: fixed in https://github.com/wso2/esb-connectors/pull/432

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/messages/getAMail.xml (hunk 4 above)
Comment (ashalya) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/messages/getAMail.xml]: fixed in https://github.com/wso2/esb-connectors/pull/432

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/threads/getAThread.xml (hunk 5 above)
Comment (ashalya) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/threads/getAThread.xml]: fixed in https://github.com/wso2/esb-connectors/pull/432

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/messages/listAllMails.xml (hunk 2 above)
Comment (ashalya) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/messages/listAllMails.xml]: fixed in https://github.com/wso2/esb-connectors/pull/432

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/messages/getAMail.xml (hunk 1 above)
Comment (ashalya) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/messages/getAMail.xml]: fixed in https://github.com/wso2/esb-connectors/pull/432

**Comment by ashalya** [comment]
All the license header changes fixed in #447

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/createMail.java [hunk 6]
```diff
@@ -33,8 +33,8 @@ public void connect(MessageContext messageContext) throws ConnectException {
...
+            encodedVal=encodedVal.replace('/', '_');
```
Comment (prabathabey) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/createMail.java]: Fix formatting issues.

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/createMail.java [hunk 7]
```diff
@@ -33,8 +33,8 @@ public void connect(MessageContext messageContext) throws ConnectException {
             String parameter = messageContext.getProperty("parameters").toString();
```
Comment (prabathabey) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/createMail.java]: Class name does not adhere the respective language standards. Please fix.

**Pull request merged**